src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -5,72 +5,61 @@
 
 import * as path from 'path';
 import * as vscode from 'vscode';
-import * as semver from 'semver';
 import { HostExecutableInformation } from '../shared/constants/hostExecutableInformation';
 import { IHostExecutableResolver } from '../shared/constants/IHostExecutableResolver';
 import { PlatformInformation } from '../shared/platform';
 import { commonOptions, languageServerOptions } from '../shared/options';
 import { existsSync } from 'fs';
 import { CSharpExtensionId } from '../constants/csharpExtensionId';
-import { getDotnetInfo } from '../shared/utils/getDotnetInfo';
 import { readFile } from 'fs/promises';
-import { RuntimeInfo } from '../shared/utils/dotnetInfo';
+import { IDotnetFindPathContext } from './dotnetRuntimeExtensionApi';
 
 export const DotNetRuntimeVersion = '8.0.10';
 
-interface IDotnetAcquireResult {
-    dotnetPath: string;
-}
-
 /**
  * Resolves the dotnet runtime for a server executable from given options and the dotnet runtime VSCode extension.
  */
 export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
     constructor(
         private platformInfo: PlatformInformation,
-        /**
-         * This is a function instead of a string because the server path can change while the extension is active (when the option changes).
-         */
-        private getServerPath: (platform: PlatformInformation) => string,
         private channel: vscode.OutputChannel,
         private extensionPath: string
     ) {}
 
     private hostInfo: HostExecutableInformation | undefined;
 
     async getHostExecutableInfo(): Promise<HostExecutableInformation> {
-        let dotnetRuntimePath = commonOptions.dotnetPath;
-        const serverPath = this.getServerPath(this.platformInfo);
-
-        // Check if we can find a valid dotnet from dotnet --version on the PATH.
-        if (!dotnetRuntimePath) {
-            const dotnetPath = await this.findDotnetFromPath();
-            if (dotnetPath) {
-                return {
-                    version: '' /* We don't need to know the version - we've already verified its high enough */,
-                    path: dotnetPath,
-                    env: this.getEnvironmentVariables(dotnetPath),
-                };
+        let dotnetExecutablePath: string;
+        if (commonOptions.dotnetPath) {
```

Comment:
this should be deprecated in favor of the options on the .net install tool side.  However I plan to tackle that in a separate PR as we'll need to separate out the O# usage of this option.

Reply:
I thought about whether we should keep the existing logic as a backup, but if this has about a mo in pre-release I'm not so concerned.